src/network/dn_comp.c

```diff
@@ -3,14 +3,26 @@
 
 /* RFC 1035 message compression */
 
+// The static functions getoffs, getlens and match are only used in this file.
+// Use checked pointers rather than bounds-safe interfaces for their parameters.
+# pragma CHECKED_SCOPE on
+
 /* label start offsets of a compressed domain name s */
-static int getoffs(short *offs, const unsigned char *base, const unsigned char *s)
+static int getoffs(_Array_ptr<short> offs : count(128), // Fixed-size, allocated in match.
+	_Nt_array_ptr<const unsigned char> base : count(0),
+	_Nt_array_ptr<const unsigned char> s : count(0))
 {
 	int i=0;
 	for (;;) {
 		while (*s & 0xc0) {
 			if ((*s & 0xc0) != 0xc0) return 0;
-			s = base + ((s[0]&0x3f)<<8 | s[1]);
+			// At this point s must have length at least 1 (s[0] cannot be 0).
```

Comment:
Why are we manually widening bounds here? The Checked C compiler can automatically widen bounds for null-terminated ptrs.

Reply:
Thanks for pointing that out. I tried to remove the bounds cast and the compiler would emit an error on out-of-bounds access for `s[1]`. It seems currently the compiler cannot automatically widen the bounds for this particular piece of code (I used the current master branch). 

I tried removing the assignment to `s`, and got the same error:
```
while ((*s & 0xc0)) {
	if ((*s & 0xc0) != 0xc0) return 0;
	base + ((s[0]&0x3f)<<8 | s[1]); // rejected
}
```

I can get rid of the error after adding a _redundant_ if guard:
```
while ((*s & 0xc0)) {
	if ((*s & 0xc0) != 0xc0) return 0;
	if (*s) base + ((s[0]&0x3f)<<8 | s[1]); // compiled
}
```

But the code below is still rejected with the assignment back to `s`:
```
while ((*s & 0xc0)) {
	if ((*s & 0xc0) != 0xc0) return 0;
	if (*s) s = base + ((s[0]&0x3f)<<8 | s[1]); // rejected
}
```

Will the analysis be improved in the future so the code snippet above can pass? Would you suggest I keep the bounds cast for now? Thanks!

P.S. I updated the code to use `_Dynamic_bounds_cast` instead. There is no need for the unchecked block now.